tensorflow/lite/micro/micro_allocator.cc

```diff
@@ -146,14 +146,16 @@ TfLiteStatus AllocationInfoBuilder::AddTensors(const SubGraph* subgraph,
 
   uint32_t operators_size = NumSubgraphOperators(subgraph);
 
-  for (size_t i = 0; i < subgraph->inputs()->size(); ++i) {
+  for (size_t i = 0;
+       subgraph->inputs() != nullptr && i < subgraph->inputs()->size(); ++i) {
```

Comment:
is it the case that some of subgraph-> inputs, outputs, etc can be nullptr but not others? or will it be the case that either all will be nullptr or none of them?

If it is the latter, having a check in the begining of the function that tests for this specific condition (and a comment explaining under what conditions that can happen) might be more readable.

Reply:
Yes, some subgraphs can have nonnull inputs / outputs. This is the case for SEANet.